src/Avalonia.Base/Threading/Dispatcher.Invoke.cs

```diff
@@ -619,4 +619,16 @@ public void Post(SendOrPostCallback action, object? arg, DispatcherPriority prio
         _ = action ?? throw new ArgumentNullException(nameof(action));
         InvokeAsyncImpl(new SendOrPostCallbackDispatcherOperation(this, priority, action, arg, true), CancellationToken.None);
     }
+
+    /// <summary>
+    /// Returns a task awaitable that would invoke continuation on specified dispatcher priority
+    /// </summary>
+    public DispatcherPriorityAwaitable AwaitWithPriority(Task task, DispatcherPriority priority) =>
```

Comment:
Interesting, can this be used like [`JoinableTaskFactory.SwitchToMainThreadAsync`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.threading.joinabletaskfactory.switchtomainthreadasync?view=visualstudiosdk-2022) in VS? Like this [example](https://gist.github.com/AArnott/4f43caa621de28619f4f)?

Reply:
Yes, it can be used from background threads too.